File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKCS/PKCS07/RecipientEncryptedKey.cs
using System;
using System.Runtime.Serialization;

//	RecipientEncryptedKey ::= SEQUENCE {
//		rid				KeyAgreeRecipientIdentifier,
//		encryptedKey	OCTET STRING
//	}

namespace Aladdin.ASN1.ISO.PKCS.PKCS7
{
	[Serializable]
	public class RecipientEncryptedKey : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ChoiceCreator<KeyAgreeRecipientIdentifier>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<OctetString				>().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected RecipientEncryptedKey(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public RecipientEncryptedKey(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public RecipientEncryptedKey(IEncodable rid, OctetString encryptedKey) : 
			base(info, rid, encryptedKey) {}

		public IEncodable	Rid				{ get { return				 this[0]; } } 
		public OctetString	EncryptedKey	{ get { return (OctetString )this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkcs/pkcs7/RecipientEncryptedKey.java
package aladdin.asn1.iso.pkcs.pkcs7;
import aladdin.asn1.*; 
import java.io.*; 

//	RecipientEncryptedKey ::= SEQUENCE {
//		rid				KeyAgreeRecipientIdentifier,
//		encryptedKey	OCTET STRING
//	}

public final class RecipientEncryptedKey extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -7905552546133249862L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ChoiceCreator(KeyAgreeRecipientIdentifier.class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(OctetString				.class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public RecipientEncryptedKey(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public RecipientEncryptedKey(IEncodable rid, OctetString encryptedKey) 
	{
		super(info, rid, encryptedKey); 
	}
	public final IEncodable     rid			() { return				  get(0); } 
	public final OctetString	encryptedKey() { return (OctetString )get(1); }
}
